### PR TITLE
{vis}[goolf/1.5.14] libXp 1.0.2 + deps (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
depends on ~~#1317~~ (for zlib, libreadline)

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 14 out of 14 (14 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/5e0a555c2fd881074893 for a full test report.

**Comment by boegel** [comment]
@wpoely86: please review?

File: easybuild/easyconfigs/l/libxcb/libxcb-1.8-goolf-1.5.14-Python-2.7.8.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+python = 'Python'
+pyver = '2.7.8'
+versionsuffix = '-%s-%s' % (python, pyver)
+dependencies = [
+    (python, pyver),
+    ('xcb-proto', '1.7', versionsuffix),
+]
+
+preconfigopts = "rm -r aclocal.m4 configure Makefile.in ltmain.sh && ./autogen.sh && "
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libxcb/libxcb-1.8-goolf-1.5.14-Python-2.7.8.eb]: maybe add a comment why this is needed?

Comment (wpoely86) [inline easybuild/easyconfigs/l/libxcb/libxcb-1.8-goolf-1.5.14-Python-2.7.8.eb]: and add autotools to the build deps?

File: easybuild/easyconfigs/x/xcb-proto/xcb-proto-1.7-goolf-1.5.14-Python-2.7.8.eb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+version = '1.7'
+
+homepage = 'http://xcb.freedesktop.org/'
+description = """The X protocol C-language Binding (XCB) is a replacement for Xlib featuring a small footprint,
+ latency hiding, direct access to the protocol, improved threading support, and extensibility."""
+
+source_urls  = ['http://xcb.freedesktop.org/dist/']
+sources = [SOURCELOWER_TAR_GZ]
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/x/xcb-proto/xcb-proto-1.7-goolf-1.5.14-Python-2.7.8.eb]: move to above `source_urls`?

File: easybuild/easyconfigs/p/printproto/printproto-1.0.5-goolf-1.5.14.eb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+name = 'printproto'
+version = '1.0.5'
+
+homepage = 'http://xorg.freedesktop.org/'
+description = """X.org PrintProto protocol headers."""
+
+source_urls = ['http://xorg.freedesktop.org/archive/individual/proto/']
+sources = [SOURCELOWER_TAR_GZ]
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/printproto/printproto-1.0.5-goolf-1.5.14.eb]: move to above `source_urls`?

File: easybuild/easyconfigs/l/libX11/libX11-1.6.1-goolf-1.5.14.eb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+name = 'libX11'
+version = '1.6.1'
+
+homepage = "http://www.freedesktop.org/wiki/Software/xlibs"
+description = """X11 client-side library"""
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://xorg.freedesktop.org/archive/individual/lib/']
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libX11/libX11-1.6.1-goolf-1.5.14.eb]: this seems to be used quite often. Maybe make a constant for it?

File: easybuild/easyconfigs/k/kbproto/kbproto-1.0.6-goolf-1.5.14.eb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+name = 'kbproto'
+version = '1.0.6'
+
+homepage = "http://www.freedesktop.org/wiki/Software/xlibs"
+description = """X.org KBProto protocol headers."""
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://xorg.freedesktop.org/archive/individual/proto/']
```
Comment (wpoely86) [inline easybuild/easyconfigs/k/kbproto/kbproto-1.0.6-goolf-1.5.14.eb]: idem ditto as below?

File: easybuild/easyconfigs/l/libX11/libX11-1.6.1-goolf-1.5.14.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/l/libX11/libX11-1.6.1-goolf-1.5.14.eb]: this is `XORG_LIB_SOURCE`, will fix

File: easybuild/easyconfigs/k/kbproto/kbproto-1.0.6-goolf-1.5.14.eb (hunk 5 above)
Comment (boegel) [inline easybuild/easyconfigs/k/kbproto/kbproto-1.0.6-goolf-1.5.14.eb]: this is `XORG_PROTO_SOURCE`, will fix

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 15 out of 16 (16 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/ffb78405303127a761d8 for a full test report.

**Comment by boegel** [comment]
@wpoely86: please review? This is synced with #1323, but more missing deps need to be added here.

File: easybuild/easyconfigs/l/libxcb/libxcb-1.11-goolf-1.5.14-Python-2.7.9.eb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
+
+source_urls  = ['http://xcb.freedesktop.org/dist/']
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = ['libxcb-no-pthread-stubs.patch']
+
+builddependencies = [
+    ('M4', '1.4.17'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libxcb/libxcb-1.11-goolf-1.5.14-Python-2.7.9.eb]: `autotools`?

File: easybuild/easyconfigs/l/libXext/libXext-1.3.3-goolf-1.5.14.eb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+homepage = "http://www.freedesktop.org/wiki/Software/xlibs"
+description = """Common X Extensions library"""
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
+toolchainopts = {'optarch': True}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://xorg.freedesktop.org/archive/individual/lib/']
+
+dependencies = [
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXext/libXext-1.3.3-goolf-1.5.14.eb]: keep the order of deps/builddeps consistent.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 12 out of 13 (13 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/02a13bc35ad589b03b00 for a full test report.

**Comment by boegel** [comment]
@wpoely86: please recheck, previous commit was incomplete (had to do a forced update)

**Comment by wpoely86** [comment]
looks fine

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 9 out of 13 (13 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/79d65e6f85d522be2299 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/17cb88751b6f629ca5c2 for a full test report.

**Comment by boegel** [comment]
good to go in, synced with #1323, thanks for the review @wpoely86!

**Pull request merged**